collect_app/src/main/java/org/odk/collect/android/utilities/FormEntryPromptUtils.java

```diff
@@ -23,23 +23,40 @@
 import org.javarosa.core.model.data.IAnswerData;
 import org.javarosa.form.api.FormEntryPrompt;
 
+import java.math.BigDecimal;
+import java.text.DecimalFormat;
 import java.util.Date;
 
 public class FormEntryPromptUtils {
 
     public static String getAnswerText(FormEntryPrompt fep, Context context) {
         IAnswerData data = fep.getAnswerValue();
-        String text;
+        final String appearance = fep.getQuestion().getAppearanceAttr();
+
         if (data instanceof DateTimeData) {
-            text = DateTimeUtils.getDateTimeLabel((Date) data.getValue(),
-                    DateTimeUtils.getDatePickerDetails(fep.getQuestion().getAppearanceAttr()), true, context);
-        } else if (data instanceof DateData) {
-            text = DateTimeUtils.getDateTimeLabel((Date) data.getValue(),
-                    DateTimeUtils.getDatePickerDetails(fep.getQuestion().getAppearanceAttr()), false, context);
-        } else {
-            text = fep.getAnswerText();
+            return DateTimeUtils.getDateTimeLabel((Date) data.getValue(),
+                    DateTimeUtils.getDatePickerDetails(appearance), true, context);
+        }
+
+        if (data instanceof DateData) {
+            return DateTimeUtils.getDateTimeLabel((Date) data.getValue(),
+                    DateTimeUtils.getDatePickerDetails(appearance), false, context);
+        }
+
+        if (data != null && appearance != null && appearance.contains("thousands-sep")) {
```

Comment:
Maybe we should check for null in other cases above too?

Reply:
Do you mean for the `instanceOf` cases? We're ok there because `instanceOf` is always false when `null` is on the left -- https://docs.oracle.com/javase/specs/jls/se8/html/jls-15.html#jls-15.20.2